lib/rules/space-in-parens.js

```diff
@@ -115,38 +118,30 @@ module.exports = {
          * @returns {boolean} True if the token is one of the exceptions for the closer paren
          */
         function isCloserException(token) {
-            return token.type === "Punctuator" && exceptions.closers.indexOf(token.value) >= 0;
+            return exceptions.closers.includes(token.value);
         }
 
         /**
-         * Determines if an opener paren should have a missing space after it
-         * @param {Object} left The paren token
-         * @param {Object} right The token after it
-         * @returns {boolean} True if the paren should have a space
+         * Determines if the first token is a Punctuator, returns second token if not
+         * @param {Object} first The first token (which has its type checked)
+         * @param {Object} second The second token
+         * @returns {Object} Token the first token is returned if it is a punctuator, else second
          */
-        function shouldOpenerHaveSpace(left, right) {
-            if (sourceCode.isSpaceBetweenTokens(left, right)) {
-                return false;
+        function checkPunctuator(first, second) {
```

Comment:
Is this function needed? Would it make more sense to use an Array.prototype method inline with a filter function at calling sites (e.g., `const punctuator = [first, second].find(token => token.type === "Punctuator")`?

I'm not sure the name `checkPunctuator` is very clear for this function, in any case. Maybe something like `getFirstPunctuator` would make more sense?

Reply:
I've updated the function to instead accept an array of tokens as the only argument and return the  first encountered token that is a Punctuator, but I would be in favour of keeping the function since this exact same inline behavior would be repeated 9 times without the function call and would quite subtly change the order of tokens each time, which I hope is made more obvious by having a function call instead.

Thoughts?

Reply:
Removed after further consideration